unittests/savanna_misc_tests.cpp

```diff
@@ -591,18 +592,141 @@ BOOST_FIXTURE_TEST_CASE(validate_qc_requiring_finalizer_policies, savanna_cluste
 
 } FC_LOG_AND_RETHROW()
 
+static void save_blockchain_data(const std::filesystem::path& blocks_dir,
+                                 const block_id_type&         id,
+                                 const std::string&           snapshot) {
+   auto source_log_file   = blocks_dir / "blocks.log";
+   auto source_index_file = blocks_dir / "blocks.index";
+
+   std::filesystem::path test_data_path { UNITTEST_TEST_DATA_DIR };
+   auto consensus_blockchain_dir = test_data_path / "consensus_blockchain";
+   auto ref_log_file             = consensus_blockchain_dir / "blocks.log";
+   auto ref_index_file           = consensus_blockchain_dir / "blocks.index";
+   auto ref_id_file_name         = consensus_blockchain_dir / "id";
+   auto ref_snapshot_file_name   = consensus_blockchain_dir / "snapshot";
+
+   // save reference blocks log
+   std::filesystem::copy_file(source_log_file, ref_log_file, std::filesystem::        copy_options::overwrite_existing);
+   std::filesystem::copy_file(source_index_file, ref_index_file, std::filesystem::        copy_options::overwrite_existing);
+
+   // save reference block_id
+   fc::cfile ref_id_file;
+   ref_id_file.set_file_path(ref_id_file_name);
+   ref_id_file.open("wb");
+   ref_id_file.write(id.data(), id.data_size());
+   ref_id_file.close();
+
+   // save reference snapshot
+   fc::cfile snapshot_file;
+   snapshot_file.set_file_path(ref_snapshot_file_name);
+   snapshot_file.open("w");
+   snapshot_file.write(snapshot.data(), snapshot.size());
+   snapshot_file.close();
+}
+
+static std::vector<char> read_reference_file(std::string filename, const char* mode) {
+   std::filesystem::path test_data_path { UNITTEST_TEST_DATA_DIR };
+   auto consensus_blockchain_dir = test_data_path / "consensus_blockchain";
+   auto ref_file_name = consensus_blockchain_dir / filename;
+
+   fc::cfile ref_file;
+   ref_file.set_file_path(ref_file_name);
+   ref_file.open(mode);
+
+   ref_file.seek_end(0);
+   auto size = ref_file.tellp();
+
+   std::vector<char> buf;
+   buf.resize(size);
+
+   ref_file.seek(0);
+   ref_file.read(buf.data(), size);
+
+   ref_file.close();
+
+   return buf;
+}
```

Comment:
You could use `fc::read_file_contents()` here, as I learned last week from Kevin!

Reply:
Thanks!